### DISCUSSION TITLE
aioble handle random disconnection

### DISCUSSION THREAD

**Discussion by puppet13th** [micropython/micropython#12849] [ESP32]
I have been handling random ble disconnection using this method, when running ble as central:
```py
try:
    async with await device.connect() as connection:
        do something...
        
except asyncio.TimeoutError:
    handle connect TimeoutError

except aioble.DeviceDisconnectedError:
    handle disconnected error

except OSError as exc:
    if exc.args == -128:
        handle disconnected error
    else:
        raise
```

sometime there is message in repl something like this:
`[aioble] E: Disconnect -128`

should aioble raise aioble.DeviceDisconnectedError instead of OSError( -128 )?

**Comment by vencax** [comment]
Have you found the reason of this Error?

**Comment by seankoji** [comment]
**#719 closed** — smartctl-exporter warnings are expected behavior for the known-bad sata4 drive (2 pending sectors, tracked in #673). Prometheus confirms all 6 drives are reporting metrics correctly; `exit status 64` is smartcltls bit flag for "device error log contains records" — not a read failure. No fix needed pending drive replacement.
